bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
-   if @dir.include? "../"
-    @dir = File.join(path, @dir, name)
-   else
-    @dir = File.join(@dir, name)
-   end
-  else
-   puts "The config file does not contain a path for the setting: #{setting}"
-   if setting == 'jobdir' then exit 1 end
+#Logger Module
+module EventLogger
+ class Log
+  include ConfigFile
+
+  #Configure the logger
+  def initialize
+    @logger = Logger.new(config.logfile)
+    @logger.level = Logger::DEBUG
+    @logger.formatter = proc do |severity, datetime, progname, msg|
+     date_format = datetime.strftime("%d-%m-%Y %H:%M:%S")
+     "[#{date_format}]  #{severity.ljust(5)}: #{msg}\n"
+    end
+    
+    #Redirect STDERR messages and append to the log file
+    $stderr.reopen(config.logfile, "a")
+  end
+  
+  #This does something
+  def method_missing(m,*args,&block)
+    log.send(m,*args,&block)
+  end
+  
+  #Return the logger
+  def log
+   @logger
   end
  end
  
- #Check to see if a given setting exists in the file
- def exists?(setting)
-  if @config[setting] != nil then return true end
+ #Returns/Creates the logger
+ def self.log
+   @log ||= Log::new
  end
+
+ def log
+  EventLogger::log
+ end 
 end
 
 class Job
- def initialize(jobdir, cmdscript, name, command, log)
-  @jobdir = jobdir
-  @cmdscript = cmdscript
-  @job = name
-  @command = command
-  @event = log
- end
+ class State
+  include EventLogger
+  include ConfigFile
+
+  #Toggles value in state hash
+  def toggle(name) 
+   @sf = YAML.load_file(config.statefile)
+
+   #Finds all true/false values in hash
+   t = @sf[name].key(true)
+   f = @sf[name].key(false)
+ 
+   #Toggles the values
+   @sf[name][:"#{t}"] = false
+   @sf[name][:"#{f}"] = true
+
+   File.open(config.statefile, 'w') { |f| YAML.dump(@sf, f) }
+  end
  
+  #Checks the current state of the requested script for the given job
+  def status(name, command)
+   @sf = YAML.load_file(config.statefile)
+   #The script will run if it's state is currently false
+   if @sf[name][:"#{command}"] == false
+    toggle(name)
+   else
+    log.error "Can't execute #{command} for #{name} as it is already set to true"
+    exit 1
+   end
+  end
+ end
+
+ include EventLogger
+ include ConfigFile
+ attr_accessor :command
+ attr_accessor :host
+
+ def initialize(name)
+   @name = name
+   @state = State.new 
+ end
+
  #Performs validation on entered command
  def validate!
-  @event.log.info "Attempting to run #{@command} script for #{@job}"
+  @script = File.join(config.jobdir(@name), @command + '.sh')
+  if @host.empty?
+   log.info "Attempting to run #{@command} script for #{@name}"
+  else 
+   log.info "Attempting to run #{@command} script for #{@name} on #{@host}"
+   host_exist?
+  end
+  
+  #Check if the directoy and file exist
   if dir_exist? and file_exist? then
    return true
   else
-   @event.log.error "Failed to validate"
+   log.error "Failed to validate"
    exit 1
   end
  end
  
+ #Checks if the host exists to run the script on
+ def host_exist?
```

Comment:
We don't really need to validate if a host exists, this command will fail if the host doesnt exists in exactly the same way as just attempting to run the script anyway so theres nothing really gained.

Reply:
Makes sense, host validation no longer exists